OpenSSL/Session.hsc

```diff
@@ -642,11 +642,17 @@ sslIOInner loc f ptr nbytes ssl
 -- | Try to read the given number of bytes from an SSL connection. On EOF an
 --   empty ByteString is returned. If the connection dies without a graceful
 --   SSL shutdown, an exception is raised.
+--
+-- NOTE: The returned bytestring could be shorter than the size requested, see:
+-- https://www.openssl.org/docs/man3.0/man3/SSL_read.html
```

Comment:
I don't see from the `tryRead` code how it can be shorter. It can be either empty or `nBytes`.

Docs says that `SSL_read_ex` will return the number of bytes read. But here the `SSL_read` is used, not `_ex`.

Have you seen instances of `read` returning strings of `0 < length < nBytes`?

Reply:
Oh, sorry, I misread `tryRead` it returns `n` from `SSL_read` result, not `nBytes`. Thank you for the PR. Merged.